### DISCUSSION TITLE
Cordova iOS: JW Player CDN Fails to Load Due to app:// Scheme

### DISCUSSION THREAD

**Discussion by saraganesh** [apache/cordova#514] [Q&A]
Hello Everyone !!

**Environemnt Details:**
       cordova-ios: 7.1.1
       
   **Issue:**
   
The app attempts to load the following JavaScript file:

    app://ssl.p.jwpcdn.com/player/v/8.36.3/jwplayer.core.controls.html5.js

However, it fails with a 404 Not Found error, preventing the JW Player from functioning correctly.

**Observations:**

- On Android, the same URL loads correctly with the https:// protocol (e.g., https://ssl.p.jwpcdn.com/player/v/8.36.3/jwplayer.core.controls.html5.js), and the video plays without issues.

- On iOS, the protocol defaults to app://, and the resource cannot be accessed.

**What I’ve Tried:**

- **Changing the Scheme in config.xml:**

        <preference name="scheme" value="https" />
        <preference name="hostname" value="localhost" />

  Unfortunately, it still defaults to app:// on iOS.

- **Using cordova-plugin-ionic-webview:**

     This changes the protocol to ionic://, but the issue persists, as the URL is still inaccessible.

- **Verified the URL:**

  The JW Player CDN (https://ssl.p.jwpcdn.com) works correctly in browsers and on Android, so the issue seems specific to the iOS WebView configuration.
  
**Questions:**

1. Is there a way to configure Cordova iOS to use the https:// protocol for external resources like the JW Player CDN?
2. How can I override the default app:// scheme in the WebView?
3. Are there any plugins or additional configurations that can help resolve this issue?

Any guidance on how to properly configure the WebView to load external resources using the https:// protocol would be greatly appreciated. 
Thank you in advance!!

![simulator_screenshot_E8AE7D01-97B4-4D69-95D1-D16A10CFAA9E](https://github.com/user-attachments/assets/7683ac26-0807-432d-bcc7-b78ffcbec4d4)

**Comment by erisu** [comment]
I believe the issue your seeing is this: https://github.com/jwplayer/jwplayer/issues/3561

It sounds like you might need to configure the `base` parameter. I suspect it is starting with `//` by default which will take the scheme that is used. It works for Android because its scheme is `https` while iOS is `app`.

You can not use HTTP or HTTPS scheme in iOS.
